packages/ogmios/src/ogmiosToCore/tx.ts

```diff
@@ -249,15 +249,22 @@ const mapTxIn = (txIn: Schema.TxIn): Cardano.TxIn => ({
   txId: Cardano.TransactionId(txIn.txId)
 });
 
-const mapDatum = (datum: Schema.TxOut['datum']) => {
+const mapInlineDatum = (datum: Schema.TxOut['datum']) => {
+  if (!datum) return;
+  if (typeof datum === 'string') return Cardano.util.HexBlob(datum);
+  if (typeof datum === 'object') return Cardano.util.HexBlob(JSON.stringify(datum));
```

Comment:
Is this correct? :thinking: Need a test with ogmios inline datum

Reply:
Do we have test data when ogmios datum is an `object`?

Reply:
Maybe it's only relevant for `datum_hash`: https://github.com/CardanoSolutions/ogmios/blob/master/clients/TypeScript/packages/schema/src/index.ts#L1120

My guess is that key = scripthash and value=datumhash there

Reply:
I think our current implementation is incorrect. Key and value there needs to be interpreted separately:
- key=hash32 (script hash? not sure)
- value=plutus_data (the hexblob)

https://github.com/input-output-hk/cardano-ledger/blob/master/eras/babbage/test-suite/cddl-files/babbage.cddl#L421

Let's try to figure out what is the difference between 'Datum' and key-value object: https://github.com/CardanoSolutions/ogmios/blob/master/clients/TypeScript/packages/schema/src/index.ts#L1120-L1123

Reply:
In that case we should drop the key and only use the value?

Or can a single output have multiple datums? :thinking: In this case we'd need to update our core type